### PR TITLE
DOC: Add HowTo for configuring IP address with ConnMan

### PR DESCRIPTION
**Pull request by dbkinder**
howtos/ip-address-config.rst
  describe how to use ConnMan to configure a static or dynanic IP addresses
  for a WiFi or Ethernet connection

howtos/howtos.rst
  added ip-address-config to list of technical notes

[skip ci]

Signed-off-by: David Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/howtos/ip-address-config.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+The Ostro Project uses Connection Manager (ConnMan), a Linux daemon and command-line utility, for managing internet connections 
+within embedded devices. 
+
+ConnMan supports these network technologies:
+
+* Ethernet, using a built-in card, connected via host or client side USB
+* WiFi, using wpa_supplicant
+* Celluar connectivity using oFono
+* Bluetooth connectivity using Bluez
```
Comment (gvancuts) [inline doc/howtos/ip-address-config.rst]: It's spelled BlueZ (capital Z)

File: doc/howtos/ip-address-config.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+     $ connmanctl config <service> –ipv4 manual <ip address> <netmask> <gateway> 
+
+   For example, using the service name discovered above: ::
+
+     $ connmanctl config ethernet_0008a209b525_cable –ipv4 manual 192.168.1.4 255.255.255.0 192.168.1.1
+
+You can assign a static IP to a discovered WiFi adapter service in the same manner.
+
+
+Configuring a Dynamic (DHCP) IP Address
```
Comment (gvancuts) [inline doc/howtos/ip-address-config.rst]: I think it'd be more appropriate to call this section something like "Connecting to a WiFi network".

Comment (gvancuts) [inline doc/howtos/ip-address-config.rst]: Or add a note in the paragraph below on how to use DHCP with Ethernet (it's basically the default behaviour so enabling ethernet is all that's needed)

Comment (uhofemei) [inline doc/howtos/ip-address-config.rst]: I can look into this more on Monday when I get back into the office. I don't know what the default configuration of Connman in Ostro looks like. The default behavior of Connman upstream is to have all technologies disabled by default for security purposes.

**Comment by gvancuts** [comment]
:+1:

File: doc/howtos/ip-address-config.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
+.. _ip-address-config:
+
+Configuring an IP Address in the Ostro |trade| OS
+######################################################
+
+The Ostro Project uses Connection Manager (ConnMan), a Linux daemon and command-line utility, for managing internet connections 
+within embedded devices.
```
Comment (barbieri) [inline doc/howtos/ip-address-config.rst]: Although correct I'd rephrase to put some emphasis on the DBus API and that it is official way to talk to connman. Then you can follow using connmanctl, then we add some DBus call that does the same, people can then use Python DBus or something like that in real products instead of trying to call system() on connmanctl

Soletta and nodejs will have nice C/FBP/JS version, so we can refer to them later.

File: doc/howtos/ip-address-config.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+Here are the steps to connect to a wireless network using DHCP:
+
+#. Open connmanctl in interactive mode, enable WiFi (if you haven't already), and scan for new WiFi services (available SSIDs)::
+
+     $ connmanctl
+     $ connmanctl> enable wifi 
+     $ connmanctl> scan wifi
+     $ connmanctl> services 
+
+#. Take note of the wireless network name and the service name displayed by the ``services`` command.  To connect to a secure
```
Comment (barbieri) [inline doc/howtos/ip-address-config.rst]: Agent is very important concept, add more emphasis and explain it is used by the daemon to call back an application when attention or input is needed. In the command line tool, available in development images, once enabled it will work until you quit the interactive session. In real life products the application is expected to always keep it alive.

File: doc/howtos/ip-address-config.rst (hunk 3 above)
Comment (dbkinder) [inline doc/howtos/ip-address-config.rst]: I think you're saying that there's a programmatic way to configure IP addresses, and while true, the immediate need was to document how to do this manually.  I'm sure there are plenty of code examples (like this) we need to write about explaining how to do things, that would be appropriate for an application programmer's cookbook with complete examples.

Comment (gvancuts) [inline doc/howtos/ip-address-config.rst]: I'd suggest we add a line to explain that this documents explains how to connect your device to a network 'manually' and _just_ mention that for proper integration in an app, people should look at the D-Bus API (and its bindings). Would that work for both of you?

**Comment by dbkinder** [comment]
force pushed updates to address Gustavo and Geoffroy's comments.  Programmatic (d-bus API) ip address configuration out of scope for this document, but should be covered in an application developer's guide/cookbook (TBD)

File: doc/howtos/ip-address-config.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+* WiFi, using wpa_supplicant
+* Celluar connectivity using oFono
+* Bluetooth connectivity using BlueZ
+
+By default all technologies are disabled at the very first startup
+to prevent unwanted wireless or wired communication from happening. ConnMan remembers the state of network
+technologies over reboots. 
+(You can read more about ConnMan at https://01.org/connman/documentation.)
+
+This tech note explains how to connect your device to a network manually using the ``conmannctl`` command line client.  For proper integration in
```
Comment (mythi) [inline doc/howtos/ip-address-config.rst]: conmannctl -> connmanctl

File: doc/howtos/ip-address-config.rst [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+* Celluar connectivity using oFono
+* Bluetooth connectivity using BlueZ
+
+By default all technologies are disabled at the very first startup
+to prevent unwanted wireless or wired communication from happening. ConnMan remembers the state of network
+technologies over reboots. 
+(You can read more about ConnMan at https://01.org/connman/documentation.)
+
+This tech note explains how to connect your device to a network manually using the ``conmannctl`` command line client.  For proper integration in 
+an application, the D-Bus service API (and its bindings) should be used.
```
Comment (gvancuts) [inline doc/howtos/ip-address-config.rst]: A couple of nitpicks:
- Shouldn't command line be spelled with an hyphen, i.e. command-line?
- Technically speaking, we'd expect people to use more than just ConnMan service API for proper integration, they'd also use its Manager API, Session API, etc. so I'd suggest we just say "... the D-Bus API (and its bindings) should be used."

**Comment by barbieri** [comment]
fair points, current document looks good to me. Later when we add connman in soletta and node.js we can link to those documents from here.

**Comment by dbkinder** [comment]
forced push to address comments
